dynect/convenient_client.go

```diff
@@ -61,10 +62,23 @@ func (c *ConvenientClient) PublishZone(zone string) error {
 	data := &PublishZoneBlock{
 		Publish: true,
 	}
-
-	if err := c.Do("PUT", url, data, nil); err != nil {
+	resp := &PublishZoneResponseBlock{}
+	if err := c.Do("PUT", url, data, resp); err != nil {
 		return fmt.Errorf("Failed to publish zone: %s", err)
 	}
+	url = fmt.Sprintf("Task/%s/", resp.Data.TaskID)
+	respTask := &TaskStateResponse{}
```

Comment:
```suggestion
	var respTask TaskStateResponse
```

If you could also move this into the `for` loop, just so it's more-tightly scoped, that would be greatly appreciated.